### PR TITLE
Implemented a poll based wait impl for non-child processes

### PR DESCRIPTION
**Pull request by diptanu**
Introduces a Wait utility method which polls a process to check whether it's still alive or not. This is useful when the process which needs to be tested is not a child process of the caller.

### CODE REVIEW

**Comment by mitchellh** [comment]
This looks great! I'm going to add some tests, but otherwise perfect.

File: wait_windows.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package plugin
+
+import (
+	"os"
+	"syscall"
+	"time"
+)
+
+const STILL_ACTIVE = 259
```
Comment (sethvargo) [inline wait_windows.go]: This doesn't appear to match Go's recommendations for constant naming.

**Comment by mitchellh** [comment]
A few updates I made to style:

* Pull the platform-independent stuff into `process.go` and call other platform-specific functions in `process_PLATFORM.go`, which is more idiomatic.

* Pull more into constants for Windows. @sethvargo it isn't idiomatic Go style but it matches the MSDN docs directly for what the returned constant is so we'll go with it.

* Made the constants for Windows unexported, don't want people somehow inadvertently seeing them in docs.

**Comment by mitchellh** [comment]
Merging!

**Pull request merged**